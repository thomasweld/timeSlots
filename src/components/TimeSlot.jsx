import React from 'react';
import PropTypes from 'prop-types';

const TimeSlot = ({ hour, available, onClickTimeSlot, name, phoneNumber }) => (
  <li
    onClick={onClickTimeSlot}
    style={{
      backgroundColor: available ? 'red' : 'none'
    }}
  >
    {hour}
    {!available && ` - Booked by ${name}`}
  </li>
);

TimeSlot.propTypes = {
  hour: PropTypes.number.isRequired,
  available: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  phoneNumber: PropTypes.string.isRequired
};

export default TimeSlot;
