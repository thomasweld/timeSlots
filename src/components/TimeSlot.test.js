import React from 'react';
import renderer from 'react-test-renderer';
import TimeSlot from './TimeSlot';

let timeSlot = {
  hour: 1,
  available: false,
  name: 'John',
  phoneNumber: '1234566543'
};

// Snapshot test example
describe('TimeSlot', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<TimeSlot {...timeSlot} onClick={() => {}} />);
    expect(tree).toMatchSnapshot();
  });
});
