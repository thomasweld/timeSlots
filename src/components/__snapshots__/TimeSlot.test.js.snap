// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeSlot should match snapshot 1`] = `
<li
  onClick={undefined}
  style={
    Object {
      "backgroundColor": "none",
    }
  }
>
  1
   - Booked by John
</li>
`;
